test/blackbox-tests/test-cases/pkg/git-repo.t

```diff
@@ -98,3 +98,38 @@ that by hashing with the checksum that we expected in the OPAM file
 
   $ cmp -s dune.lock/foo.files/$FILES_NAME "$FILES_FOLDER/$FILES_NAME" && echo "The contents match"
   The contents match
+
+The git repos support repo should also be able to handle unusual objects in our tree, which can e.g. happen with submodules.
```

Comment:
This paragraph seems a bit long. Wrapping it to 80 chars would make it a little nicer.